addons/account/models/account_journal.py

```diff
@@ -310,7 +310,7 @@ def _compute_selected_payment_method_codes(self):
         These will be then used to display or not payment method specific fields in the view.
         """
         for journal in self:
-            codes = [line.code for line in journal.inbound_payment_method_line_ids + journal.outbound_payment_method_line_ids]
+            codes = [line.code for line in journal.inbound_payment_method_line_ids + journal.outbound_payment_method_line_ids if line.code]
```

Comment:
`code` is a required field on `account.payment.method` so this doesn't make any sense to me